fideslib/oauth/api/deps.py

```diff
@@ -0,0 +1,59 @@
+from typing import Generator
+
+from fastapi import Depends, Security
+from fastapi.security import SecurityScopes
+from sqlalchemy.orm import Session
+
+from fideslib.core.config import FidesConfig
+from fideslib.core.config import get_config as core_get_config
+from fideslib.db.session import get_db_session
+from fideslib.models.client import ClientDetail
+from fideslib.oauth.api.urn_registry import TOKEN
+from fideslib.oauth.oauth_util import verify_oauth_client as verify
+from fideslib.oauth.schemas.oauth import OAuth2ClientCredentialsBearer
+
+
+def get_config() -> FidesConfig:
+    """Returns the config settings.
+
+    This should be overridden by the installing package.
+    """
+    return core_get_config()
+
+
+def get_db() -> Generator[Session, None, None]:
+    """Return our database session.
+
+    This should be overridden by the installing package.
+    """
+    try:
+        SessionLocal = get_db_session(core_get_config())
+        db = SessionLocal()
+        yield db
+    finally:
+        db.close()
+
+
+def oauth2_scheme() -> OAuth2ClientCredentialsBearer:
+    """Creates the oauth2 scheme from the token."""
+    return OAuth2ClientCredentialsBearer(tokenUrl=TOKEN)
```

Comment:
I didn't do the v1 prefix because with this being an installable library I thought it could make things weird. fidesops and fidesctl could be on different api versions so I didn't want to force things here. Since this will be a dependency injection the installing package can override this. I am going to add the note to override the dependency to the comment to make it clear.